### PR TITLE
Make `AndroidAdvertisement` parcelable

### LINKED ISSUE #610: Feature request: Parcelable Advertisement on Android
I wonder if you would consider making the `AndroidAdvertisement` implement `Parcelable`?

The [Android native ScanResult](https://developer.android.com/reference/android/bluetooth/le/ScanResult) implements the Parcelable interface, so I think this should be possible, though I don't know if there are internal details of the Advertisement implementation that make it more difficult.

The use case is an android view that shows a list of devices in range, implemented using Jetpack Compose. I would like to be able to seamlessly handle app configuration changes using `rememberSaveable`.

Yes, I could store the list of advertisements in a global singleton, outside of the view hierarchy, but that feels less in line with Android architectural idioms.

### PR DESCRIPTION
**Pull request by twyatt**
Closes #610.

### CODE REVIEW

**Comment by cedrickcooke** [comment]
Any reason you're not just using the [parcelize](https://developer.android.com/kotlin/parcelize) plugin?

**Comment by twyatt** [comment]
> Any reason you're not just using the [parcelize](https://developer.android.com/kotlin/parcelize) plugin?

🤦 I forgot about that plugin! TY!
I'll update the PR.

File: core/src/androidMain/kotlin/AndroidAdvertisement.kt [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+import android.os.Parcelable
+
 @Deprecated(
     message = "Moved as nested class of `AndroidAdvertisement`.",
     replaceWith = ReplaceWith("AndroidAdvertisement.BondState"),
 )
 public typealias BondState = AndroidAdvertisement.BondState
 
-public interface AndroidAdvertisement : Advertisement {
+public interface AndroidAdvertisement : Advertisement, Parcelable {
```
Comment (QuantumRand) [inline core/src/androidMain/kotlin/AndroidAdvertisement.kt:11]: This makes this a `major` change, correct?

**Review by QuantumRand — APPROVED** [review]

File: core/src/androidMain/kotlin/AndroidAdvertisement.kt (hunk 1 above)
Comment (twyatt) [reply core/src/androidMain/kotlin/AndroidAdvertisement.kt:11]: Thank you; yes.

**Review by cedrickcooke — APPROVED** [review]

**Review by davidtaylor-juul — APPROVED** [review]

**Pull request merged**